multiqc/modules/hicpro/hicpro.py

```diff
@@ -0,0 +1,392 @@
+        keys['Dangling_end_pairs'] = { 'color': '#ff5c33', 'name': 'Same Fragment - Dangling Ends' }
+        keys['Religation_pairs'] = { 'color': '#cc2900', 'name': 'Re-ligation' }
+        keys['Dumped_pairs'] = { 'color': '#661400', 'name': 'Dumped pairs' }
+
+        # Config for the plot
+        config = {
+            'id': 'hicpro_filtering_plot',
+            'title': 'HiC-Pro: Filtering Statistics',
+            'ylab': '# Read Pairs',
+            'cpswitch_counts_label': 'Number of Read Pairs'
+        }
+        
+        return bargraph.plot(self.hicpro_data, keys, config)
+
+    def hicpro_contact_chart (self):
+        """ Generate the HiC-Pro interaction plot """
+
+        # Specify the order of the different possible categories
+        keys = OrderedDict()
+        keys['cis_shortRange'] = { 'color': '#0039e6', 'name': 'Unique: cis <= 20Kbp' }
+        keys['cis_longRange'] = { 'color': '#809fff', 'name': 'Unique: cis > 20Kbp' }
+        keys['trans_interaction']  = { 'color': '#009933', 'name': 'Unique: trans' }
+        keys['duplicates'] = { 'color': '#a9a2a2', 'name': 'Duplicate read pairs' }
+        
+        # Config for the plot
+        config = {
+            'id': 'hicpro_contact_plot',
+            'title': 'HiC-Pro: Contact Statistics',
+            'ylab': '# Pairs',
+            'cpswitch_counts_label': 'Number of Pairs'
+        }
+        
+        return bargraph.plot(self.hicpro_data, keys, config)
+  
+    def hicpro_as_chart (self):
+        """ Generate Allele-specific plot"""
+        
+        keys = OrderedDict()
+        keys['Valid_pairs_from_ref_genome_(1-1)'] = { 'color': '#e6550d', 'name': 'Genome1 specific read pairs (1-1)' }
+        keys['Valid_pairs_from_ref_genome_with_one_unassigned_mate_(0-1/1-0)'] = { 'color': '#fdae6b', 'name': 'Genome1 with one unassigned mate (0-1/1-0)' }
+        keys['Valid_pairs_from_alt_genome_(2-2)']  = { 'color': '#756bb1', 'name': 'Genome2 specific read pairs (2-2)' }
+        keys['Valid_pairs_from_alt_genome_with_one_unassigned_mate_(0-2/2-0)'] = { 'color': '#bcbddc', 'name': 'Genome2 with one unassigned mate (0-2/2-0)' }
+        keys['Valid_pairs_from_alt_and_ref_genome_(1-2/2-1)'] = { 'color': '#a6611a', 'name': 'Trans homologuous read pairs (1-2/2/1)' }
+        keys['Valid_pairs_with_both_unassigned_mated_(0-0)'] = { 'color': '#cccccc', 'name': 'Unassigned read pairs' }
+        keys['Valid_pairs_with_at_least_one_conflicting_mate_(3-)'] = { 'color': '#a9a2a2', 'name': 'Conflicting read pairs' }
+
+        # Config for the plot
+        config = {
+            'id': 'hicpro_asan_plot',
+            'title': 'HiC-Pro: Allele-specific Statistics',
+            'ylab': '# Pairs',
+            'cpswitch_counts_label': 'Number of Pairs'
+        }
+
+        return bargraph.plot(self.hicpro_data, keys, config)
+
+                
+    def hicpro_capture_chart (self):
```

Comment:
Please remove these for now. Can always add in at a later date. I'm actually kind of surprised that this doesn't trigger an error..!

Reply:
ok